test/trento/domain/sap_system/sap_system_test.exs

```diff
@@ -4202,6 +4208,361 @@ defmodule Trento.SapSystemTest do
+          sid: sid,
+          ensa_version: ensa_version
+        )
+      ]
+
+      assert_events_and_state(
+        initial_events,
+        [
+          %RegisterDatabaseInstance{
+            sap_system_id: sap_system_id,
+            host_id: host_id,
+            instance_number: absent_db_instance_number,
+            health: :passing
+          },
+          %RegisterDatabaseInstance{
+            sap_system_id: sap_system_id,
+            host_id: host_id,
+            instance_number: present_db_instance_number,
+            health: :passing
+          },
+          %RegisterApplicationInstance{
+            sap_system_id: sap_system_id,
+            host_id: host_id,
+            instance_number: absent_app_instance_number,
+            health: :passing,
+            ensa_version: ensa_version
+          },
+          %RegisterApplicationInstance{
+            sap_system_id: sap_system_id,
+            host_id: host_id,
+            instance_number: present_app_instance_number,
+            health: :passing,
+            ensa_version: ensa_version
+          }
+        ],
+        [
+          %DatabaseInstanceMarkedPresent{
+            instance_number: absent_db_instance_number,
+            host_id: host_id,
+            sap_system_id: sap_system_id
+          },
+          %ApplicationInstanceMarkedPresent{
+            instance_number: absent_app_instance_number,
+            host_id: host_id,
+            sap_system_id: sap_system_id
+          }
+        ],
+        fn state ->
+          assert %SapSystem{
+                   sid: ^sid,
+                   ensa_version: ^ensa_version,
+                   application: %SapSystem.Application{
+                     sid: ^sid,
+                     instances: [
+                       %SapSystem.Instance{
+                         absent_at: nil
+                       },
+                       %SapSystem.Instance{
+                         absent_at: nil
+                       }
+                     ]
+                   },
+                   database: %SapSystem.Database{
+                     sid: ^sid,
+                     instances: [
+                       %SapSystem.Instance{
+                         absent_at: nil
+                       },
+                       %SapSystem.Instance{
+                         absent_at: nil
+                       }
+                     ]
+                   }
+                 } = state
+        end
+      )
+    end
+
+    test "absent instances receiving 'mark as absent' command should mark as absent" do
+      sap_system_id = Faker.UUID.v4()
+      sid = fake_sid()
+      ensa_version = EnsaVersion.ensa1()
+      host_id = Faker.UUID.v4()
+      absent_db_instance_number = "01"
+      present_db_instance_number = "02"
+      absent_app_instance_number = "03"
+      present_app_instance_number = "04"
+      absent_db_absent_at = DateTime.utc_now()
+      absent_app_absent_at = DateTime.utc_now()
+
+      initial_events = [
```

Comment:
Aren't we mixing to many things here?
Wouldn't be better to have one tests for the databases and other for the applications?
After all they are totally different code and flows?